### PR TITLE
CT-2917 update the case details page

### PR DESCRIPTION
**Pull request by njpreddy**
## Description
CT-2917 Update the case details section

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [x] (6) No TODO's without new ticket numbers
* [x] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Screenshots
Case 1 - Third party requests data to be sent to the data subject

Case1

Case 2 - Third party requests data for themselves

Case2

Case 3 - Data subject requesting their own data

Case3

Case 4 - Data subject requesting data to be sent to third party

Case4

### Related JIRA tickets
Jira ticket CT-2917
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-2917

### Manual testing instructions
Create the following cases with the appropriate steps:

87925944-ebd96280-ca78-11ea-877f-41163baaeca1

### CODE REVIEW

File: app/helpers/cases_helper.rb
Comment (willmcb) [inline app/helpers/cases_helper.rb:332]: Might be worth extracting these out into the `en.yml`

File: app/views/cases/offender_sar/_case_details.html.slim
Comment (willmcb) [inline app/views/cases/offender_sar/_case_details.html.slim:22]: I wonder if it might be clearer to wrap this in a method that explains what this mean so we can have:

```ruby
unless data_subject_is_requesting_date_for_themselves?
```
I think this may make it easier to reason about 👍

**Review by willmcb — COMMENTED** [review]
Looks good 👍 - I have a few suggestions - let me know if you'd like to chat about them

File: app/helpers/cases_helper.rb
Comment (njpreddy) [reply app/helpers/cases_helper.rb:332]: Great spot Will, done :-)

File: app/views/cases/offender_sar/_case_details.html.slim
Comment (Polsonby) [reply app/views/cases/offender_sar/_case_details.html.slim:22]: As we just discussed, Nick, wondering if we might just get away with simpler logic anyway?

``` 
- if case_details.third_party?
   = render partial 'third_party_requester_details'
```
... because "third_party" specifically means "third party requester" for Offender SAR cases and is slightly independent of the recipient - recipient will either be subject or requester if "third_party" is true but I don't _think_ we need this part of the page to check

And then in the next section about where to send the data, we can just use 

```
- if case_details.third_party_recipient?
          = render partial: 'cases/offender_sar/third_party_recipient_details'
```

File: app/views/cases/offender_sar/_request_details.html.slim
Comment (Polsonby) [inline app/views/cases/offender_sar/_request_details.html.slim:5]: There's a crash on this partial that Alex reported if the optional "request_dated" date field is nil. Please could you amend this line to:

```suggestion
    = l(case_details.request_dated, format: :default) if case_details.request_dated.present?
```

**Review by Polsonby — COMMENTED** [review]
Looks great! I wasn't sure about nuking all the "one liner" view partials for each field, since that's how the other correspondence types are displayed but I really like the way the offender sar view files have ended up when you look at the files side by side in the folder, I think we're in a nice place with this PR :) 

There are a few tiny things we could potentially consider tweaking, but not sure I'd hold up merging for them! 

Happy to approve either way - just wondering if you might be able to make one change to request dated as [per comment above](https://github.com/ministryofjustice/correspondence_tool_staff/pull/1205#discussion_r458413703) though?

File: app/views/cases/offender_sar/_case_details.html.slim
Comment (Polsonby) [inline app/views/cases/offender_sar/_case_details.html.slim:17]: I _think_ we can just do 
```suggestion
        - if case_details.third_party?
```
- you can't ever have third_party requester AND third party recipient so it's redundant to test for it here
image

File: app/views/cases/offender_sar/_case_details.html.slim
Comment (Polsonby) [inline app/views/cases/offender_sar/_case_details.html.slim:22]: We don't need to check for third party requester here, I think? Also, the enums for the recipient generate free predicate methods, so we can write the above as simply 
```suggestion
        - if case_details.third_party_recipient?
```

File: app/views/cases/offender_sar/_case_details.html.slim
Comment (Polsonby) [inline app/views/cases/offender_sar/_case_details.html.slim:15]: I'm just wondering about the partial name here - is this the requester details section?  Or is it just part of it? This uncertainty is what I hoped we'd resolve by naming the partials using the same names as the steps they relate to :)

File: app/helpers/cases_helper.rb
Comment (Polsonby) [inline app/helpers/cases_helper.rb:341]: Could we move this into the offender_decorator, maybe? 

Also, wondering if we can get rid of this conditional by using the same approach as we did in the offender_decorator for `subject_type_display` if we wire up the keys in the en.yml file using the same strings as the values for recipient here. 

Compare with 
image

So we'd have keys in the yaml file like `helpers.label.offender_sar.recipient.requester_recipient` or more simply `helpers.label.offender_sar.requester_recipient` and then the method becomes something like 

```suggestion
def get_sar_recipient_label(recipient)
  I18n.t('helpers.label.offender_sar.recipient.' + recipient)
end
```
(... but using the equivalent form for a decorator i.e. `object.recipient`)

File: app/views/cases/offender_sar/_request_details.html.slim
Comment (njpreddy) [reply app/views/cases/offender_sar/_request_details.html.slim]: Done!

File: app/views/cases/offender_sar/_case_details.html.slim
Comment (njpreddy) [reply app/views/cases/offender_sar/_case_details.html.slim]: Done!

File: app/views/cases/offender_sar/_case_details.html.slim
Comment (njpreddy) [reply app/views/cases/offender_sar/_case_details.html.slim]: Done!

**Review by Polsonby — APPROVED** [review]
Nice work :)

**Pull request merged**
